routes/users.js

```diff
@@ -0,0 +1,84 @@
+import express from 'express';
+import bcrypt from 'bcryptjs';
+
+import { addUserValidationRules, validateData } from '../validate.js';
+import { User } from '../models';
+
+const router = express.Router();
+
+const getUsers = async (request, response) => {
+  try {
+    const users = await User.findAll();
+    response.status(200).send(users);
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+const addUser = async (request, response) => {
+  const { userName, email, firstName, lastName, password } = request.body;
+
+  try {
+    const salt = await bcrypt.genSalt(10);
+    const hashPassword = await bcrypt.hash(password, salt);
+    await User.create({
+      firstName,
+      lastName,
+      email,
+      password: hashPassword,
+      userName,
+    });
+    response.status(201).json({
```

Comment:
It is safer to return this function call so you will stop the execution of the call here when you have an resposne. Here you wont have any side efects but if you will have multiple branches you could run into some unwanted side effects.

Reply:
Please make sure you handle this everywhere.